project-2-e-commerce/src/component/cart/index.js

```diff
@@ -1,26 +1,33 @@
-import { createListNav } from '../../helpers'
+import './cart.scss'
+import { createListNav, getLocalData, formatPrice, getTotal } from '../../helpers'
 import { useTranslation } from 'react-i18next'
 import NavComponent from '../nav'
 import { Link, useLocation } from 'react-router-dom'
-import { useDispatch, useSelector } from 'react-redux'
-import './cart.scss'
+import { useDispatch } from 'react-redux'
 import CartItem from './CartItem'
-import { formatPrice, getTotal } from '../../helpers'
-import { useState } from 'react'
-import { deleteUserCart } from '../../store/slices/UserSlice'
+import { useEffect, useState } from 'react'
+import { setCartsLength, updateUserCart } from '../../store/slices/UserSlice'
 
 function Cart() {
     const dispatch = useDispatch()
     const { t } = useTranslation()
     const { pathname } = useLocation()
-    const { isLogged, userData } = useSelector(({user}) => user)
-    const localCarts = JSON.parse(localStorage.getItem('local-carts'))
-    const cartList = isLogged ? userData.cart : localCarts ? localCarts : []
+    const { localCarts, isLogged, userData } = getLocalData()
+    const cartList = isLogged ? userData.carts : localCarts ? localCarts : []
     const totalPay = getTotal(cartList)
     const [isOpenConfirmModal, setIsOpenConfirmModal] = useState(false)
     const [idCartDelete, setIdCartDelete] = useState('')
     const [isChange, setIschange] = useState(false)
 
+    useEffect(()=>{
+        return () => {
+            if (isLogged) {
+                const { carts: cartsUpdate, id: userId } = JSON.parse(localStorage.getItem('user-login'))
+                dispatch(updateUserCart({userId, carts: {carts: cartsUpdate}}))
+            }
+        }
+    })
```

Comment:
```suggestion
    useEffect(()=>{
         if (isLogged) {
             const { carts: cartsUpdate, id: userId } = JSON.parse(localStorage.getItem('user-login'))
             dispatch(updateUserCart({userId, carts: {carts: cartsUpdate}}))
         }
    })
```
Xem lại cái action updateUserCart. Sao lại phải lưu như thế này nhỉ?
`carts: {carts: cartsUpdate}`
sao không phải là
```
const { carts, id: userId } = JSON.parse(localStorage.getItem('user-login'))
dispatch(updateUserCart({ userId, carts }))
```